common/app/rendering/core/JavascriptProps.scala

```diff
@@ -0,0 +1,21 @@
+package rendering.core
+
+import conf.Configuration
+import play.api.libs.json.{JsValue, Json, Writes}
+
+case class JavascriptPropsConfig(
+  bundleUrl: String = conf.Static("javascripts/ui.bundle.browser.js"),
+  polyfillioUrl: String =
+    if (conf.switches.Switches.PolyfillIO.isSwitchedOn) common.Assets.js.polyfillioUrl
+    else conf.Static("javascripts/vendor/polyfillio.fallback.js"),
+  beaconUrl: String = Configuration.debug.beaconUrl
+)
+case class JavascriptProps(config: JavascriptPropsConfig){
+  def asJsValue(): JsValue = Json.toJson(this)
```

Comment:
In Scala you can define parameterless methods with or without parenthesis. Usually using parenthesis is done to indicate side effect.
`asJsValue` only read internal state so you can define it without the parenthesis. (Also save you 2 chars when calling it). 
Same for `default` below